src/org/rascalmpl/core/library/lang/rascalcore/check/RascalConfig.rsc

```diff
@@ -353,7 +353,8 @@ void checkOverloading(map[str,Tree] namedTrees, Solver s){
     }
     try {
     consDef = range(consNameDef);
-    for(d1 <- consDef, d2 <- consDef, d1.defined != d2.defined, t1 := s.getType(d1), t2 := s.getType(d2), t1.adt == t2.adt,
+    consTypes = (d : s.getType(d) | d <- consDef);
+    for(d1 <- consDef, d2 <- consDef, d1.defined != d2.defined, t1 := consTypes[d1], t2 := consTypes[d1], t1.adt == t2.adt,
```

Comment:
There's a typo in `t2 := consTypes[d1]`, t2 should be defined by the constypes of d2 ?

Reply:
Might have to run the measurements again.

Reply:
Also if this loop is hot then introducing types for the variable bindings will really speed things up. It removes the need to traverse the scope stack for finding bound variables with the same name.

Reply:
ah sorry, I did it in the browser, my bad, I ran the test with the correct edit.